server/controllers/donations/addDonation.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable camelcase */
+import { NextFunction, Response } from 'express';
+import { Donation } from '../../database/models';
+import { CustomedError } from '../../utils';
+import { paramsSchema, donationSchema } from '../../utils/validation';
+
+const addDonation = async (req: any, res: Response, next: NextFunction) => {
+  const donorId: number = req.user.id;
```

Comment:
- you can destructure the id here and rename it, and add this line inside the try